### PR TITLE
Fix/set equipment date

### PR DESCRIPTION
**Pull request by dabick14**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Screenshots (if appropriate):

### CODE REVIEW

File: api/src/resolvers/campaigns/campaigns-cypher.ts [hunk 1]
```diff
@@ -2,7 +2,11 @@ export const SetEquipmentDeadline = `
 MATCH (gatheringService:GatheringService {id: $id})
 SET gatheringService.equipmentStartDate = date($startDate),
     gatheringService.equipmentEndDate = date($endDate)
-MERGE (equipmentDate:TimeGraph:EquipmentDate {date:date($startDate)})
+MERGE (equipmentDate:TimeGraph {date:date($startDate)})
+ON MATCH 
+SET equipmentDate:EquipmentDate
+ON CREATE
+SET equipmentDate:EquipmentDate
```
Comment (jaedag) [inline api/src/resolvers/campaigns/campaigns-cypher.ts:9]: You do realise that to have a condition, and then to run the exact same code in both cases, is tautology right? 

You should remove this ON MATCH and ON CREATE 

I don't think they are solving any problem

File: api/src/resolvers/campaigns/campaigns-resolver.ts [hunk 2]
```diff
@@ -110,17 +110,17 @@ export const campaignsMutation = {
       const startDate = new Date(equipmentCampaign.campaign?.equipmentStartDate)
       const newStartDate = new Date(args.startDate)
 
+      const setEquipmentDuration = rearrangeCypherObject(
+        await session.run(SetEquipmentDeadline, args)
```
Comment (jaedag) [inline api/src/resolvers/campaigns/campaigns-resolver.ts:114]: no error handling here

**Review by jaedag — CHANGES_REQUESTED** [review]

File: api/src/resolvers/campaigns/campaigns-resolver.ts (hunk 2 above)
Comment (dabick14) [reply api/src/resolvers/campaigns/campaigns-resolver.ts:114]: It's wrapped in a try catch. The try starts earlier

File: api/src/resolvers/campaigns/campaigns-cypher.ts (hunk 1 above)
Comment (dabick14) [reply api/src/resolvers/campaigns/campaigns-cypher.ts:6]: Seen. Changing it

**Review by jaedag — APPROVED** [review]

**Review by ahadziii — APPROVED** [review]

**Pull request merged**